# This configuration file specifies the location and 
# orientation of thrusting motors for a multirotor
# aircraft. Note that servos are not included, only
# motors that produce thrust and torque to satisfy 
# control commands.

num_motors: 4

# Vector from the center of mass to the plane of 
# rotation for each motor in the body frame (in meters).
# ROSflight firmware supports up to 10 motor outputs.
motor_positions:
  - [0.2298, 0.2298, 0]
  - [-0.2298, 0.2298, 0]
  - [-0.2298, -0.2298, 0]
  - [0.2298, -0.2298, 0]

# Unit vectors specifying the direction of thrust for 
# each motor. 
# Example: [0,0,-1] for a motor aligned (in the 
# negative direction) with the body-z frame.
motor_orientations:
  - [0, 0, -1]
  - [0, 0, -1]
  - [0, 0, -1]
  - [0, 0, -1]

# Scalar that determines the direction of rotation
# of the propeller. Used to calculate the torque 
# generated by the spinning propeller.
# d = 1 if rotation is CCW
# d = -1 if rotation is CW
motor_directions:
  - -1
  - 1
  - -1
  - 1

# Misc motor/frame parameters
vehicle_mass: 3.5   # kg
motor_resistance: 0.085   # ohms
air_density: 1.225        # kg/m3
motor_kv: 0.02894         # SI units
no_load_current: 1.01     # amps
prop_diameter: 0.381      # meters
max_voltage: 25.0         # Volts
prop_ct: 0.075
prop_cq: 0.0045
